server/data/events.ts

```diff
@@ -0,0 +1,224 @@
+import collections from '../config/mongoCollections';
+import { ObjectId } from 'mongodb';
+import { ApplicationObject, ContactObject, CycleObject, EventObject, MediaObject, UserObject } from '../typings';
+
+export default {
+    /**
+     * Get event by id
+     * @param {ObjectId} cycleId
+     * @param {ObjectId} applicationId
+     * @param {ObjectId} eventId
+     * @returns {Promise<object>} The event given the ids above
+     */
+    readById: async (cycleId: ObjectId, applicationId: ObjectId, eventId: ObjectId):Promise<object> => {
+
+        const cyclesCollection: any = await collections.cycles();
+        const cycle = await cyclesCollection.findOne({_id: new ObjectId(cycleId)});
+        if(cycle === null) throw new Error('Could not find cycle');
+
+        let application = null;
+        for(let element of cycle.applications){
+            if(element._id.toString() === applicationId.toString()){
+                application = element;
+            }
+        }
+        if(application === null) throw new Error('Could not find application');
+
+        let event = null;
+        for(let element of application.events){
+            if(element._id.toString() === eventId.toString()){
+                event = element
+            }
+        }
+        if(event === null) throw new Error('Could not find event');
+
+        event._id = event._id.toString();
+        return event;
+    },
+
+    /**
+     * Get all events
+     * @param {ObjectId} cycleId
+     * @param {ObjectId} applicationId
+     * @returns {Promise<object[]>}All the events of the application
+     */
+    readAll: async (cycleId: ObjectId, applicationId: ObjectId): Promise<object[]> => {
+
+        const cyclesCollection: any = await collections.cycles();
+        const cycle = await cyclesCollection.findOne({_id: new ObjectId(cycleId)});
+        if(cycle === null) throw new Error('Could not find cycle');
+
+        let application = null;
+        for(let element of cycle.applications){
+            if(element._id.toString() === applicationId.toString()){
+                application = element;
+                break;
+            }
+        }
+        if(application === null) throw new Error('Could not find application');
+        
+        let events = application.events.map((element) => element.toString());
+
+        return events
+    },
+
+    /**
+     * Create event
+     * @param {ObjectId} cycleId 
+     * @param {string} title 
+     * @param {Date} date 
+     * @param {String} location 
+     * @returns {Promise<object>} if the event was created, throws otherwise 
+     */
+    create: async (cycleId: ObjectId, applicationId: ObjectId, title: string, date: Date, location: string): Promise<object> => {
```

Comment:
JS doc does not match actual function. Also cycleId, applicationId, etc will be passed as strings and not as ObjectIds. They get converted to those types on this layer (data)

Reply:
Also cycleId is not gonna get provided, they can only add events for the users current cycle. Instead the routes will be giving you the current userId and based on that you grab their current cycle and handle from there.

Reply:
Agree with everything besides the cycleId not being passed. There is no indication in the users collection, or cycles collection for the users current cycle.